src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,143 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private Object[] internalArray = new Object[10];
+    private int size = 0;
+
     @Override
     public void add(T value) {
-
+        if (size == internalArray.length) {
+            int newLength = (int) (internalArray.length * 1.5);
+            Object[] internalArrayTemp = new Object[newLength];
+            System.arraycopy(internalArray,0, internalArrayTemp, 0, internalArray.length);
+            internalArray = internalArrayTemp;
+        }
+        internalArray[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index > size || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("Index doesn't exist, size of array = "
+                    + size);
+        }
+        if (size == internalArray.length) {
```

Comment:
Create private method grow() with logic of array growing
[Don’t create repeating code](https://mate-academy.github.io/jv-program-common-mistakes/java-core/collections/array-list.html#dont-create-repeating-code)